hy/models.py

```diff
@@ -339,13 +341,26 @@ def replace(self, other, recursive=True):
             self.conversion = other.conversion
         return self
 
+    def __repr__(self):
+        return 'hy.models.FComponent({})'.format(
+            super(Object, self).__repr__() +
+            ', conversion=' + repr(self.conversion))
+
 class FString(Sequence):
     """
     Generic Hy F-String object, for smarter f-string handling.
     Mimics ast.JoinedStr, but using String and FComponent.
     """
     def __new__(cls, s=None, brackets=None):
-        value = super().__new__(cls, s)
+        value = super().__new__(cls,
```

Comment:
this is overly complicated and can be simplified and made more explicit with something like
```python
import operator
value = super().__new__(cls,
  # Join adjacent string nodes for the sake of equality
  # testing.
      tuple(
          reduce(operator.add, components) if is_string else next(components)
          for is_string, components in groupby(s, lambda x: isinstance(x, String))
      )
      if s else s)
```

Reply:
Also `Sequence` is a child of `tuple` and can take a generator so the call to `tuple` can be removed and left as a generator to make 

```python
(
    reduce(operator.add, components) if is_string else next(components)
    for is_string, components in groupby(s, lambda x: isinstance(x, String))
)
```

Reply:
Smart using `next` like that. Done, except I kept the short variable names for the sake of line length.

Reply:
please use the descriptive names please, it doesn't break 90 characters and it'll make this so much easier to parse and understand for the next person who has to look at this.

Reply:
I think 90 characters is a chonker of a line. I try to keep lines to like 70 characters past the indent. Anyway, I restored the names and added some more spacing. I hate Python's inside-out ternary operator.

Reply:
Good catch, thanks. Done.